### PR TITLE
Proposal: export emotion-theming from react-emotion

### PR DESCRIPTION
**Pull request by evan-scott-zocdoc**
**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Code complete

While refactoring for a proof of concept, it quickly became cumbersome to have to split up the import just to pull in theming HOCs.

Since theming functionality is partially built-in by default in emotion v8 via supporting `p => p.theme` etc, I think it makes sense to go the rest of the way and re-export `emotion-theming` for ease of use.

~~It might make sense to get rid of `emotion-theming` entirely and just have it be part of `react-emotion`?~~ in v9 maybe

### CODE REVIEW

**Comment by evan-scott-zocdoc** [comment]
Side note, I don't use typescript so no idea what the errors mean. I think think I have everything else working...

**Comment by evan-scott-zocdoc** [comment]
@tkh44 What do you think? I'm @probablyup's work account btw :)

**Comment by tkh44** [comment]
I like it on the surface, but what does this do the bundle size? Previously we were just importing one file from emotion-theming.

**Comment by evan-scott-zocdoc** [comment]
It increases by ~.76kB.

**Comment by evan-scott-zocdoc** [comment]
I guess the question here is how often do emotion users also use emotion-theming? If the majority do not also use theming, then this could not make sense.

That being said, treeshaking and DCE should take care of pieces of the library that aren't used right?

**Comment by tkh44** [comment]
I think the majority of users are using theming in some way. I'm leaning towards merging this.

File: packages/react-emotion/types/index.d.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { StatelessComponent, ComponentClass, CSSProperties } from 'react';
 import { Interpolation as EmotionInterpolation } from 'emotion';
 
 export * from 'emotion';
+export * from 'emotion-theming';
```
Comment (tkh44) [inline packages/react-emotion/types/index.d.ts]: This should be updated to match @mitchellhamilton's work in #457

**Comment by mitchellhamilton** [comment]
I agree, I'm also going to try some stuff with babel to see if I can get the size down a bit as well because I checked and right now tree shaking won't work with what babel 6 outputs but with babel 7 it should work

**Comment by evan-scott-zocdoc** [comment]
Kk I'll rebase

File: packages/react-emotion/types/index.d.ts (hunk 1 above)
Comment (mitchellhamilton) [reply packages/react-emotion/types/index.d.ts]: I don't understand, #457 shouldn't require any changes to this PR.

**Comment by evan-scott-zocdoc** [comment]
Do you folks prefer re-exporting `emotion-theming`, or simply merging it into `react-emotion`?

**Comment by evan-scott-zocdoc** [comment]
I suppose merging it would be a breaking change, nvm.

**Comment by evan-scott-zocdoc** [comment]
@mitchellhamilton Could also add this in preparation for webpack 4: https://github.com/webpack/webpack/tree/next/examples/side-effects

File: packages/react-emotion/package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
   },
   "dependencies": {
     "babel-plugin-emotion": "^8.0.10",
+    "emotion-theming": "^8.0.10",
```
Comment (mitchellhamilton) [inline packages/react-emotion/package.json]: This also needs to be added to `preact-emotion`. Also, in the rollup config, in the `preact-emotion` bit, `prop-types` and `emotion-theming` need to added to `external`.

File: packages/react-emotion/types/index.d.ts (hunk 1 above)
Comment (tkh44) [reply packages/react-emotion/types/index.d.ts]: Oops, I thought you manually changed the export in that PR. Didn't realize the update fixed it.

File: packages/react-emotion/package.json (hunk 2 above)
Comment (evan-scott-zocdoc) [reply packages/react-emotion/package.json]: done

**Comment by probablyup** [comment]
Redoing this under my personal account

**Pull request closed without merging**